### DISCUSSION TITLE
Can I modify the population after I loaded the vcf file into tree sequence ?

### DISCUSSION THREAD

**Discussion by silvewheat** [tskit-dev/tskit#2173] [Q&A] [answered]
Hello,

Is there any way to modify the population information after I loaded the vcf file using the following code (the Tutorial of tsinfer)?
Because I find some individuals should belong to another population after I calculate the GNN(genealogical nearest neighbour). 
Or sometimes I may need to split a big population into two subpopulations to calculate some statistics.
![image](https://user-images.githubusercontent.com/19344997/161470378-d5716aef-38dc-4082-ab14-ac761ebabe8a.png)

Best,
Yudong

**Comment by jeromekelleher** [comment, marked as answer]
Hi @silvewheat :wave:

You can update the population information afterwards using the [tables API](https://tskit.dev/tutorials/tables_and_editing.html). Essentially, you'll need to add a new population to the popualtion table, and then edit the node table so that the population column reflects the required structures.

However, you don't actually have to do this if all you want to do is compute GNNs. The GNN [method](https://tskit.dev/tskit/docs/stable/python-api.html#tskit.TreeSequence.genealogical_nearest_neighbours) takes a list of "sample sets", which may or may not correspond to populations.
